diff --git a/cert-append.txt b/cert-append.txt
new file mode 100644
index 0000000000..aa45f4ef3b
--- /dev/null
+++ b/cert-append.txt
@@ -0,0 +1,123 @@
+
+#
+# Certificate "mitmproxy"
+#
+# Issuer: O=mitmproxy,CN=mitmproxy
+# Serial Number:5f:28:05:f7:2b:cd:aa:b0:59:29:13:2a:a9:f6:bb:52:c6:80:81:91
+# Subject: O=mitmproxy,CN=mitmproxy
+# Not Valid Before: Thu May 23 20:54:05 2024
+# Not Valid After : Tue May 23 20:54:05 2034
+# Fingerprint (SHA-256): E1:21:80:3A:62:66:62:9A:4F:46:A1:07:67:2E:A2:3F:DD:D0:64:DC:49:55:BF:4C:6B:E6:F5:42:19:19:6A:A5
+# Fingerprint (SHA1): C7:C8:A2:F5:4A:B2:CC:2F:27:E7:9E:61:80:BB:0A:CB:C2:05:92:9B
+CKA_CLASS CK_OBJECT_CLASS CKO_CERTIFICATE
+CKA_TOKEN CK_BBOOL CK_TRUE
+CKA_PRIVATE CK_BBOOL CK_FALSE
+CKA_MODIFIABLE CK_BBOOL CK_FALSE
+CKA_LABEL UTF8 "mitmproxy"
+CKA_CERTIFICATE_TYPE CK_CERTIFICATE_TYPE CKC_X_509
+CKA_SUBJECT MULTILINE_OCTAL
+\060\050\061\022\060\020\006\003\125\004\003\014\011\155\151\164
+\155\160\162\157\170\171\061\022\060\020\006\003\125\004\012\014
+\011\155\151\164\155\160\162\157\170\171
+END
+CKA_ID UTF8 "0"
+CKA_ISSUER MULTILINE_OCTAL
+\060\050\061\022\060\020\006\003\125\004\003\014\011\155\151\164
+\155\160\162\157\170\171\061\022\060\020\006\003\125\004\012\014
+\011\155\151\164\155\160\162\157\170\171
+END
+CKA_SERIAL_NUMBER MULTILINE_OCTAL
+\002\024\137\050\005\367\053\315\252\260\131\051\023\052\251\366
+\273\122\306\200\201\221
+END
+CKA_VALUE MULTILINE_OCTAL
+\060\202\003\065\060\202\002\035\240\003\002\001\002\002\024\137
+\050\005\367\053\315\252\260\131\051\023\052\251\366\273\122\306
+\200\201\221\060\015\006\011\052\206\110\206\367\015\001\001\013
+\005\000\060\050\061\022\060\020\006\003\125\004\003\014\011\155
+\151\164\155\160\162\157\170\171\061\022\060\020\006\003\125\004
+\012\014\011\155\151\164\155\160\162\157\170\171\060\036\027\015
+\062\064\060\065\062\063\062\060\065\064\060\065\132\027\015\063
+\064\060\065\062\063\062\060\065\064\060\065\132\060\050\061\022
+\060\020\006\003\125\004\003\014\011\155\151\164\155\160\162\157
+\170\171\061\022\060\020\006\003\125\004\012\014\011\155\151\164
+\155\160\162\157\170\171\060\202\001\042\060\015\006\011\052\206
+\110\206\367\015\001\001\001\005\000\003\202\001\017\000\060\202
+\001\012\002\202\001\001\000\237\250\324\100\007\357\252\312\214
+\103\371\136\324\100\047\246\216\133\163\053\300\000\074\313\343
+\011\342\074\311\043\016\052\105\033\071\002\212\375\362\075\127
+\240\272\007\365\361\146\175\135\126\252\344\310\032\302\245\145
+\133\162\277\372\130\170\213\161\122\017\042\162\213\135\257\021
+\057\121\044\071\012\007\333\264\302\216\175\065\076\132\167\266
+\057\055\014\026\005\227\065\056\170\367\127\122\065\052\004\325
+\220\110\107\125\255\132\213\102\203\042\337\177\127\201\206\207
+\174\220\224\217\075\224\106\013\216\005\243\135\272\262\021\121
+\301\325\204\231\156\040\273\022\351\374\352\271\205\262\133\023
+\132\066\202\357\275\063\165\207\250\322\214\213\123\147\017\171
+\330\174\251\151\071\156\075\330\233\276\364\303\330\144\317\277
+\144\071\153\260\165\356\221\323\306\367\056\061\364\010\304\241
+\135\175\020\340\223\045\325\347\045\005\162\305\036\131\262\261
+\021\306\175\212\014\222\273\255\270\135\134\307\114\103\074\170
+\334\261\121\267\206\051\062\071\134\367\111\276\302\206\106\171
+\002\170\077\044\003\130\213\002\003\001\000\001\243\127\060\125
+\060\017\006\003\125\035\023\001\001\377\004\005\060\003\001\001
+\377\060\023\006\003\125\035\045\004\014\060\012\006\010\053\006
+\001\005\005\007\003\001\060\016\006\003\125\035\017\001\001\377
+\004\004\003\002\001\006\060\035\006\003\125\035\016\004\026\004
+\024\066\032\273\206\012\126\021\156\344\331\076\026\362\147\277
+\212\305\317\244\170\060\015\006\011\052\206\110\206\367\015\001
+\001\013\005\000\003\202\001\001\000\166\345\342\013\374\016\054
+\237\375\013\332\064\034\046\271\150\331\124\207\006\357\056\153
+\304\166\007\344\022\226\317\346\371\341\161\202\006\306\106\227
+\376\132\053\122\322\212\166\365\315\211\305\270\070\203\055\207
+\277\316\275\245\215\023\006\217\266\110\234\220\107\341\100\074
+\333\160\364\162\111\262\015\136\214\331\071\340\115\225\300\371
+\341\334\325\172\100\346\151\062\345\344\351\342\243\015\344\132
+\332\242\242\244\050\037\152\003\131\051\214\316\273\031\020\240
+\077\264\125\337\204\310\345\363\056\212\325\253\043\043\263\233
+\203\335\261\347\144\345\327\157\150\100\363\076\054\117\341\162
+\021\003\261\163\367\130\004\033\020\303\304\061\227\066\153\235
+\047\254\000\210\156\162\233\147\321\353\205\234\056\133\354\220
+\230\156\376\226\203\370\103\032\107\246\002\113\367\174\017\106
+\135\361\133\173\204\274\374\325\307\336\163\316\112\213\217\342
+\226\200\061\151\376\304\320\164\157\345\067\142\160\030\162\004
+\160\016\040\047\234\157\247\044\110\231\237\136\373\135\363\377
+\124\327\335\337\167\147\153\360\211
+END
+CKA_NSS_MOZILLA_CA_POLICY CK_BBOOL CK_TRUE
+CKA_NSS_SERVER_DISTRUST_AFTER CK_BBOOL CK_FALSE
+CKA_NSS_EMAIL_DISTRUST_AFTER CK_BBOOL CK_FALSE
+
+# Trust for "mitmproxy"
+# Issuer: O=mitmproxy,CN=mitmproxy
+# Serial Number:5f:28:05:f7:2b:cd:aa:b0:59:29:13:2a:a9:f6:bb:52:c6:80:81:91
+# Subject: O=mitmproxy,CN=mitmproxy
+# Not Valid Before: Thu May 23 20:54:05 2024
+# Not Valid After : Tue May 23 20:54:05 2034
+# Fingerprint (SHA-256): E1:21:80:3A:62:66:62:9A:4F:46:A1:07:67:2E:A2:3F:DD:D0:64:DC:49:55:BF:4C:6B:E6:F5:42:19:19:6A:A5
+# Fingerprint (SHA1): C7:C8:A2:F5:4A:B2:CC:2F:27:E7:9E:61:80:BB:0A:CB:C2:05:92:9B
+CKA_CLASS CK_OBJECT_CLASS CKO_NSS_TRUST
+CKA_TOKEN CK_BBOOL CK_TRUE
+CKA_PRIVATE CK_BBOOL CK_FALSE
+CKA_MODIFIABLE CK_BBOOL CK_FALSE
+CKA_LABEL UTF8 "mitmproxy"
+CKA_CERT_SHA1_HASH MULTILINE_OCTAL
+\307\310\242\365\112\262\314\057\047\347\236\141\200\273\012\313
+\302\005\222\233
+END
+CKA_CERT_MD5_HASH MULTILINE_OCTAL
+\042\162\171\114\176\017\145\366\355\271\126\322\023\105\356\100
+END
+CKA_ISSUER MULTILINE_OCTAL
+\060\050\061\022\060\020\006\003\125\004\003\014\011\155\151\164
+\155\160\162\157\170\171\061\022\060\020\006\003\125\004\012\014
+\011\155\151\164\155\160\162\157\170\171
+END
+CKA_SERIAL_NUMBER MULTILINE_OCTAL
+\002\024\137\050\005\367\053\315\252\260\131\051\023\052\251\366
+\273\122\306\200\201\221
+END
+CKA_TRUST_SERVER_AUTH CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_EMAIL_PROTECTION CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_CODE_SIGNING CK_TRUST CKT_NSS_TRUSTED_DELEGATOR
+CKA_TRUST_STEP_UP_APPROVED CK_BBOOL CK_FALSE
diff --git a/clean-append-files.sh b/clean-append-files.sh
new file mode 100644
index 0000000000..692fdd0f4a
--- /dev/null
+++ b/clean-append-files.sh
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+
+function clean-append-files
+{
+	sed '/\/\*\* \[SECTION\] ISOLATION/{$!{:begin N; s/\/\*\* \[SECTION\] ISOLATION.*\n\n//; t end b begin :end}}' $1 \
+	| sed '/\/\*\* \[SECTION\] SANITIZING/{$!{:begin N; s/\/\*\* \[SECTION\] SANITIZING.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] CACHE/{$!{:begin N; s/\/\*\* \[SECTION\] CACHE.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] HISTORY/{$!{:begin N; s/\/\*\* \[SECTION\] HISTORY.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] REFERERS/{$!{:begin N; s/\/\*\* \[SECTION\] REFERERS.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] RFP/{$!{:begin N; s/\/\*\* \[SECTION\] RFP.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] WEBGL/{$!{:begin N; s/\/\*\* \[SECTION\] WEBGL.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] AUTOPLAY/{$!{:begin N; s/\/\*\* \[SECTION\] AUTOPLAY.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] CONTAINERS/{$!{:begin N; s/\/\*\* \[SECTION\] CONTAINERS.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[SECTION\] BRANDING/{$!{:begin N; s/\/\*\* \[SECTION\] BRANDING.*\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[CATEGORY\] WINDOWS/{$!{:begin N; s/\/\*\* \[CATEGORY\] WINDOWS.*\n\n\n//; t end b begin :end}}' \
+	| sed '/\/\*\* \[CATEGORY\] OVERRIDES/{$!{:begin N; s/\/\*\* \[CATEGORY\] OVERRIDES.*\n\n//; t end b begin :end}}' \
+	| sed -n 's/^[[:blank:]]*pref("/pref("/p' \
+	| sed 's/).*$/);/g' \
+	| sed 's/",[[:blank:]]*/", /g' \
+	| tac \
+	| sort -t ' ' -k 1,1 -u  \
+	| sed '/javascript.use_us_english_locale/d' \
+	| sed '/intl.accept_languages/d'
+}
\ No newline at end of file
diff --git a/generate-append-firefoxjs.sh b/generate-append-firefoxjs.sh
new file mode 100644
index 0000000000..f10756e131
--- /dev/null
+++ b/generate-append-firefoxjs.sh
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+function generate-append-firefoxjs
+{
+	sed -n 's/^[[:blank:]]*pref("//p' $1 \
+	| sed 's/".*$//g' \
+	| xargs -I % -P 0 grep \"%\" $2 \
+	| sort -t ' ' -k 1,1 -u
+}
\ No newline at end of file
diff --git a/librewolf.js b/librewolf.js
new file mode 100644
index 0000000000..b2e174f085
--- /dev/null
+++ b/librewolf.js
@@ -0,0 +1,549 @@
+
+/** LIBREWOLF SETTINGS
+ * 
+ * take the time to read and understand, but also to customize the settings to find your own setup.
+ * the answers to the most common questions can be found at https://librewolf.net/docs/faq/.
+ * 
+ * WARNING: make sure the first line of this file is empty. this is a known bug.
+ */
+pref("librewolf.cfg.version", "7.10");
+
+
+/** INDEX
+ * the file is organized in categories, and each one has a number of sections:
+ * 
+ * PRIVACY [ISOLATION, SANITIZING, CACHE AND STORAGE, HISTORY AND SESSION RESTORE, QUERY STRIPPING]
+ * NETWORKING [HTTPS, REFERERS, WEBRTC, PROXY, DNS, PREFETCHING AND SPECULATIVE CONNECTIONS]
+ * FINGERPRINTING [RFP, WEBGL]
+ * SECURITY [SITE ISOLATION, CERTIFICATES, TLS/SSL, PERMISSIONS, SAFE BROWSING, OTHERS]
+ * REGION [LOCATION, LANGUAGE]
+ * BEHAVIOR [DRM, SEARCH AND URLBAR, DOWNLOADS, AUTOPLAY, POP-UPS AND WINDOWS, MOUSE]
+ * EXTENSIONS [USER INSTALLED, SYSTEM, EXTENSION FIREWALL]
+ * BUILT-IN FEATURES [UPDATER, SYNC, LOCKWISE, CONTAINERS, DEVTOOLS, SHOPPING, OTHERS]
+ * UI [BRANDING, HANDLERS, FIRST LAUNCH, NEW TAB PAGE, ABOUT, RECOMMENDED]
+ * TELEMETRY
+ * WINDOWS [UPDATES, OTHERS]
+ */
+
+/** [CATEGORY] PRIVACY */
+
+/** [SECTION] ISOLATION
+ * default to strict mode, which includes:
+ * 1. dFPI for both normal and private windows
+ * 2. strict blocking lists for trackers
+ * 3. shims to avoid breakage caused by blocking lists
+ * 4. stricter policies for xorigin referrers
+ * 5. dFPI specific cookie cleaning mechanism
+ * 6. query stripping
+ * 
+ * the desired category must be set with pref() otherwise it won't stick.
+ * the UI that allows to change mode manually is hidden.
+ */
+pref("browser.contentblocking.category", "strict");
+// enable APS
+pref("privacy.partition.always_partition_third_party_non_cookie_storage", true);
+pref("privacy.partition.always_partition_third_party_non_cookie_storage.exempt_sessionstorage", false);
+
+/** [SECTION] SANITIZING 
+ * all the cleaning prefs true by default except for siteSetting and offlineApps,
+ * which is what we want. users should set manual exceptions in the UI if there
+ * are cookies they want to keep.
+ */
+pref("privacy.clearOnShutdown.offlineApps", true);
+pref("privacy.sanitize.sanitizeOnShutdown", true);
+pref("privacy.sanitize.timeSpan", 0);
+
+/** [SECTION] CACHE AND STORAGE */
+pref("browser.cache.disk.enable", false); // disable disk cache
+/** prevent media cache from being written to disk in pb, but increase max cache size to avoid playback issues */
+pref("browser.privatebrowsing.forceMediaMemoryCache", true);
+pref("media.memory_cache_max_size", 65536);
+pref("browser.shell.shortcutFavicons", false); // disable favicons in profile folder
+pref("browser.helperApps.deleteTempFileOnExit", true); // delete temporary files opened with external apps
+
+/** [SECTION] HISTORY AND SESSION RESTORE
+ * since we hide the UI for modes other than custom we want to reset it for
+ * everyone. same thing for always on PB mode.
+ */
+pref("privacy.history.custom", true);
+pref("browser.privatebrowsing.autostart", false);
+pref("browser.formfill.enable", false); // disable form history
+pref("browser.sessionstore.privacy_level", 2); // prevent websites from storing session data like cookies and forms
+
+/** [SECTION] QUERY STRIPPING
+ * currently we set the same query stripping list that brave uses:
+ * https://github.com/brave/brave-core/blob/f337a47cf84211807035581a9f609853752a32fb/browser/net/brave_site_hacks_network_delegate_helper.cc#L29
+ */
+pref("privacy.query_stripping.strip_list", "__hsfp __hssc __hstc __s _hsenc _openstat dclid fbclid gbraid gclid hsCtaTracking igshid mc_eid ml_subscriber ml_subscriber_hash msclkid oft_c oft_ck oft_d oft_id oft_ids oft_k oft_lk oft_sk oly_anon_id oly_enc_id rb_clickid s_cid twclid vero_conv vero_id wbraid wickedid yclid");
+/**
+ * librewolf specific pref that allows to include the query stripping lists in uBO by default.
+ * the asset file is fetched every 7 days.
+ */
+pref("librewolf.uBO.assetsBootstrapLocation", "https://gitlab.com/librewolf-community/browser/source/-/raw/main/assets/uBOAssets.json");
+
+/** [SECTION] LOGGING
+ * these prefs are off by default in the official Mozilla builds,
+ * so it only makes sense that we also disable them.
+ * See https://gitlab.com/librewolf-community/settings/-/issues/240
+ */
+pref("browser.dom.window.dump.enabled", false);
+pref("devtools.console.stdout.chrome", false);
+
+
+
+/** [CATEGORY] NETWORKING */
+
+/** [SECTION] HTTPS */
+pref("dom.security.https_only_mode", true); // only allow https in all windows, including private browsing
+pref("network.auth.subresource-http-auth-allow", 1); // block HTTP authentication credential dialogs
+
+/** [SECTION] REFERERS
+ * to enhance privacy but keep a certain level of usability we trim cross-origin
+ * referers to only send scheme, host and port, instead of completely avoid sending them.
+ * as a general rule, the behavior of referes which are not cross-origin should not
+ * be changed.
+ */
+pref("network.http.referer.XOriginTrimmingPolicy", 2);
+
+/** [SECTION] WEBRTC
+ * there is no point in disabling webrtc as mDNS protects the private IP on linux, osx and win10+.
+ * the private IP address is only used in trusted environments, eg. allowed camera and mic access.
+ */
+pref("media.peerconnection.ice.default_address_only", true); // use a single interface for ICE candidates, the vpn one when a vpn is used
+
+/** [SECTION] PROXY */
+pref("network.gio.supported-protocols", ""); // disable gio as it could bypass proxy
+pref("network.file.disable_unc_paths", true); // hidden, disable using uniform naming convention to prevent proxy bypass
+pref("network.proxy.socks_remote_dns", true); // forces dns query through the proxy when using one
+pref("media.peerconnection.ice.proxy_only_if_behind_proxy", true); // force webrtc inside proxy when one is used
+
+/** [SECTION] DNS */
+pref("network.dns.disablePrefetch", true); // disable dns prefetching
+pref("network.dns.skipTRR-when-parental-control-enabled", false);  // Arkenfox user.js v117
+pref("doh-rollout.provider-list", '[{"UIName":"Mozilla Cloudflare","uri":"https://mozilla.cloudflare-dns.com/dns-query"},{"UIName":"Quad9","uri":"https://dns.quad9.net/dns-query"},{"UIName":"Quad9+ECS","uri":"https://dns11.quad9.net/dns-query"}]');
+/**
+ * librewolf does not use DoH, but it can be enabled with the following prefs:
+ * pref("network.trr.mode", 2);
+ * pref("network.trr.uri", "https://dns.quad9.net/dns-query");
+ * 
+ * the possible modes are:
+ * 0 = default
+ * 1 = browser picks faster
+ * 2 = DoH with system dns fallback
+ * 3 = DoH without fallback
+ * 5 = DoH is off, default currently
+ */
+
+/** [SECTION] PREFETCHING AND SPECULATIVE CONNECTIONS
+ * disable prefecthing for different things such as links, bookmarks and predictions.
+ */
+pref("network.predictor.enabled", false);
+pref("network.prefetch-next", false);
+pref("network.http.speculative-parallel-limit", 0);
+pref("browser.places.speculativeConnect.enabled", false);
+// disable speculative connections and domain guessing from the urlbar
+pref("browser.urlbar.speculativeConnect.enabled", false);
+
+
+
+/** [CATEGORY] FINGERPRINTING */
+
+/** [SECTION] RFP
+ * librewolf should stick to RFP for fingerprinting. we should not set prefs that interfere with it
+ * and disabling API for no good reason will be counter productive, so it should also be avoided.  
+ */
+pref("privacy.resistFingerprinting", true);
+// rfp related settings
+pref("privacy.resistFingerprinting.block_mozAddonManager", true); // prevents rfp from breaking AMO
+pref("browser.display.use_system_colors", false); // default, except Win
+/**
+ * increase the size of new RFP windows for better usability, while still using a rounded value.
+ * if the screen resolution is lower it will stretch to the biggest possible rounded value.
+ * also, expose hidden letterboxing pref but do not enable it for now.
+ */
+pref("privacy.window.maxInnerWidth", 1600);
+pref("privacy.window.maxInnerHeight", 900);
+pref("privacy.resistFingerprinting.letterboxing", false);
+// this ensures there is no rounding error when computing the window size, see #1569.
+pref("browser.toolbars.bookmarks.visibility", "always");
+
+/** [SECTION] WEBGL */
+pref("webgl.disabled", true);
+
+
+
+/** [CATEGORY] SECURITY */
+
+/** [SECTION] CERTIFICATES */
+pref("security.cert_pinning.enforcement_level", 2); // enable strict public key pinning, might cause issues with AVs
+/**
+ * enable safe negotiation and show warning when it is not supported. might cause breakage
+ * if the the server does not support RFC 5746, in tha case SSL_ERROR_UNSAFE_NEGOTIATION
+ * will be shown.
+ */
+pref("security.ssl.require_safe_negotiation", true);
+pref("security.ssl.treat_unsafe_negotiation_as_broken", true);
+/**
+ * our strategy with revocation is to perform all possible checks with CRL, but when a cert
+ * cannot be checked with it we use OCSP stapled with hard-fail, to still keep privacy and
+ * increase security.
+ * crlite is in mode 3 by default, which allows us to detect false positive with OCSP.
+ * in v103, when crlite is fully mature, it will switch to mode 2 and no longer double-check.
+ */
+pref("security.remote_settings.crlite_filters.enabled", true);
+pref("security.OCSP.require", true); // set to hard-fail, might cause SEC_ERROR_OCSP_SERVER_ERROR
+
+/** [SECTION] TLS/SSL */
+pref("security.tls.enable_0rtt_data", false); // disable 0 RTT to improve tls 1.3 security
+pref("security.tls.version.enable-deprecated", false); // make TLS downgrades session only by enforcing it with pref(), default
+pref("browser.xul.error_pages.expert_bad_cert", true); // show relevant and advanced issues on warnings and error screens
+
+/** [SECTION] PERMISSIONS */
+pref("permissions.delegation.enabled", false); // force permission request to show real origin
+pref("permissions.manager.defaultsUrl", ""); // revoke special permissions for some mozilla domains
+
+/** [SECTION] SAFE BROWSING
+ * disable safe browsing, including the fetch of updates. reverting the 7 prefs below
+ * allows to perform local checks and to fetch updated lists from google.
+ */
+pref("browser.safebrowsing.malware.enabled", false);
+pref("browser.safebrowsing.phishing.enabled", false);
+pref("browser.safebrowsing.blockedURIs.enabled", false);
+pref("browser.safebrowsing.provider.google4.gethashURL", "");
+pref("browser.safebrowsing.provider.google4.updateURL", "");
+pref("browser.safebrowsing.provider.google.gethashURL", "");
+pref("browser.safebrowsing.provider.google.updateURL", "");
+/**
+ * disable safe browsing checks on downloads, both local and remote. the resetting prefs
+ * control remote checks, while the first one is for local checks only.
+ */
+pref("browser.safebrowsing.downloads.enabled", false);
+pref("browser.safebrowsing.downloads.remote.enabled", false);
+pref("browser.safebrowsing.downloads.remote.block_potentially_unwanted", false);
+pref("browser.safebrowsing.downloads.remote.block_uncommon", false);
+// empty for defense in depth
+pref("browser.safebrowsing.downloads.remote.url", "");
+pref("browser.safebrowsing.provider.google4.dataSharingURL", "");
+
+/** [SECTION] OTHERS */
+pref("network.IDN_show_punycode", true); // use punycode in idn to prevent spoofing
+pref("pdfjs.enableScripting", false); // disable js scripting in the built-in pdf reader
+
+
+
+/** [CATEGORY] REGION */
+
+/** [SECTION] LOCATION
+ * replace google with mozilla as the default geolocation provide and prevent use of OS location services
+ */
+pref("geo.provider.network.url", "https://location.services.mozilla.com/v1/geolocate?key=%MOZILLA_API_KEY%");
+pref("geo.provider.ms-windows-location", false); // [WINDOWS]
+pref("geo.provider.use_corelocation", false); // [MAC]
+pref("geo.provider.use_gpsd", false); // [LINUX]
+pref("geo.provider.use_geoclue", false); // [LINUX]
+
+/** [SECTION] LANGUAGE
+ * show language as en-US for all users, regardless of their OS language and browser language.
+ * both prefs must use pref() and not pref to work.
+ */
+pref("javascript.use_us_english_locale", true);
+pref("intl.accept_languages", "en-US, en");
+// disable region specific updates from mozilla
+pref("browser.region.network.url", "");
+pref("browser.region.update.enabled", false);
+
+
+
+/** [CATEGORY] BEHAVIOR */
+
+/** [SECTION] DRM */
+pref("media.eme.enabled", false); // master switch for drm content
+pref("media.gmp-manager.url", "data:text/plain,"); // prevent checks for plugin updates when drm is disabled
+// disable the widevine and the openh264 plugins
+pref("media.gmp-provider.enabled", false);
+pref("media.gmp-gmpopenh264.enabled", false);
+
+/** [SECTION] SEARCH AND URLBAR
+ * disable search suggestion and do not update opensearch engines.
+ */
+
+pref("browser.urlbar.suggest.searches", false);
+pref("browser.search.suggest.enabled", false);
+pref("browser.search.update", false);
+pref("browser.search.separatePrivateDefault", true); // [FF70+] // Arkenfox user.js v119
+pref("browser.search.separatePrivateDefault.ui.enabled", true); // [FF71+]  // Arkenfox user.js v119
+pref("browser.search.serpEventTelemetry.enabled",false);
+
+pref("browser.urlbar.suggest.mdn", true);
+pref("browser.urlbar.addons.featureGate", false);
+pref("browser.urlbar.mdn.featureGate", false);
+pref("browser.urlbar.pocket.featureGate", false);
+pref("browser.urlbar.trending.featureGate", false);
+pref("browser.urlbar.weather.featureGate", false);
+
+// these are from Arkenfox, I decided to put them here.
+pref("browser.download.start_downloads_in_tmp_dir", true); // Arkenfox user.js v118
+
+
+/**
+ * the pref disables the whole feature and hide it from the ui
+ * (as noted in https://bugzilla.mozilla.org/show_bug.cgi?id=1755057).
+ * this also includes the best match feature, as it is part of firefox suggest.
+ */
+pref("browser.urlbar.quicksuggest.enabled", false);
+pref("browser.urlbar.suggest.weather", false); // disable weather suggestions in urlbar once they are no longer behind feature gate
+
+// Allows the user to add a custom search engine in the settings.
+pref("browser.urlbar.update2.engineAliasRefresh", true);
+
+/** [SECTION] DOWNLOADS
+ * user interaction should always be required for downloads, as a way to enhance security by asking
+ * the user to specific a certain save location. 
+ */
+pref("browser.download.useDownloadDir", false);
+pref("browser.download.autohideButton", false); // do not hide download button automatically
+pref("browser.download.manager.addToRecentDocs", false); // do not add downloads to recents
+pref("browser.download.alwaysOpenPanel", false); // do not expand toolbar menu for every download, we already have enough interaction
+
+/** [SECTION] AUTOPLAY
+ * block autoplay unless element is right-clicked. this means background videos, videos in a different tab,
+ * or media opened while other media is played will not start automatically.
+ * thumbnails will not autoplay unless hovered. exceptions can be set from the UI.
+ */
+pref("media.autoplay.default", 5);
+
+/** [SECTION] POP-UPS AND WINDOWS
+ * prevent scripts from resizing existing windows and opening new ones, by forcing them into
+ * new tabs that can't be resized as well.
+ */
+pref("dom.disable_window_move_resize", true);
+pref("browser.link.open_newwindow", 3);
+pref("browser.link.open_newwindow.restriction", 0);
+
+/** [SECTION] MOUSE */
+pref("browser.tabs.searchclipboardfor.middleclick", false); // prevent mouse middle click on new tab button to trigger searches or page loads
+
+
+
+/** [CATEGORY] EXTENSIONS */
+
+/** [SECTION] USER INSTALLED
+ * extensions are allowed to operate on restricted domains, while their scope
+ * is set to profile+applications (https://mike.kaply.com/2012/02/21/understanding-add-on-scopes/).
+ * an installation prompt should always be displayed.
+ */
+pref("extensions.webextensions.restrictedDomains", "");
+pref("extensions.enabledScopes", 5); // hidden
+pref("extensions.postDownloadThirdPartyPrompt", false);
+/**
+ * the pref disables quarantined domains.
+ * this is a security feature, we should remove it with v116 as there will be a UI to control this per-extension.
+ * unless we patch remote settings we rely on static dumps. this means even if we did not flip this pref it would
+ * not make a difference at the moment.
+ */
+pref("extensions.quarantinedDomains.enabled", false);
+
+/** [SECTION] SYSTEM
+ * built-in extension are not allowed to auto-update. additionally the reporter extension
+ * of webcompat is disabled. urls are stripped for defense in depth.
+ */
+pref("extensions.systemAddon.update.enabled", false);
+pref("extensions.systemAddon.update.url", "");
+pref("extensions.webcompat-reporter.enabled", false);
+pref("extensions.webcompat-reporter.newIssueEndpoint", "");
+
+/** [SECTION] EXTENSION FIREWALL
+ * the firewall can be enabled with the below prefs, but it is not a sane default:
+ * pref("extensions.webextensions.base-content-security-policy", "default-src 'none'; script-src 'none'; object-src 'none';");
+ * pref("extensions.webextensions.base-content-security-policy.v3", "default-src 'none'; script-src 'none'; object-src 'none';");
+ */
+
+
+
+/** [CATEGORY] BUILT-IN FEATURES */
+
+/** [SECTION] UPDATER
+ * since we do not bake auto-updates in the browser it doesn't make sense at the moment.
+ */
+pref("app.update.auto", false);
+
+/** [SECTION] SYNC
+ * this functionality is disabled by default but it can be activated in one click.
+ * this pref fully controls the feature, including its ui.
+ */
+pref("identity.fxaccounts.enabled", false);
+
+/** [SECTION] LOCKWISE
+ * disable the default password manager built into the browser, including its autofill
+ * capabilities and formless login capture.
+ */
+pref("signon.rememberSignons", false);
+pref("signon.autofillForms", false);
+pref("extensions.formautofill.addresses.enabled", false);
+pref("extensions.formautofill.creditCards.enabled", false);
+pref("signon.formlessCapture.enabled", false);
+
+/** [SECTION] CONTAINERS
+ * enable containers and show the settings to control them in the stock ui
+ */
+pref("privacy.userContext.enabled", true);
+pref("privacy.userContext.ui.enabled", true);
+
+/** [SECTION] DEVTOOLS
+ * disable remote debugging.
+ */
+pref("devtools.debugger.remote-enabled", false); // default, but subject to branding so keep it
+pref("devtools.selfxss.count", 0); // required for devtools console to work
+
+/** [SECTION] SHOPPING
+ * disable the fakespot shopping sidebar
+ */
+pref("browser.shopping.experience2023.enabled", false);
+pref("browser.shopping.experience2023.optedIn", 2);
+pref("browser.shopping.experience2023.active", false);
+
+/** [SECTION] OTHERS */
+pref("webchannel.allowObject.urlWhitelist", ""); // remove web channel whitelist
+pref("services.settings.server", "https://%.invalid") // set the remote settings URL (REMOTE_SETTINGS_SERVER_URL in the code)
+
+
+
+/** [CATEGORY] UI */
+
+/** [SECTION] BRANDING
+ * set librewolf support and releases urls in the UI, so that users land in the proper places.
+ */
+pref("app.support.baseURL", "https://support.librewolf.net/");
+pref("browser.search.searchEnginesURL", "https://librewolf.net/docs/faq/#how-do-i-add-a-search-engine");
+pref("browser.geolocation.warning.infoURL", "https://librewolf.net/docs/faq/#how-do-i-enable-location-aware-browsing");
+pref("app.feedback.baseURL", "https://librewolf.net/#questions");
+pref("app.releaseNotesURL", "https://gitlab.com/librewolf-community/browser");
+pref("app.releaseNotesURL.aboutDialog", "https://gitlab.com/librewolf-community/browser");
+pref("app.update.url.details", "https://gitlab.com/librewolf-community/browser");
+pref("app.update.url.manual", "https://gitlab.com/librewolf-community/browser");
+
+/** [SECTION] FIRST LAUNCH
+ * disable what's new and ui tour on first start and updates. the browser
+ * should also not stress user about being the default one.
+ */
+pref("browser.startup.homepage_override.mstone", "ignore");
+pref("startup.homepage_override_url", "about:blank");
+pref("startup.homepage_welcome_url", "about:blank");
+pref("startup.homepage_welcome_url.additional", "");
+pref("browser.messaging-system.whatsNewPanel.enabled", false);
+pref("browser.uitour.enabled", false);
+pref("browser.uitour.url", "");
+pref("browser.shell.checkDefaultBrowser", false);
+
+/** [SECTION] NEW TAB PAGE
+ * we want NTP to display nothing but the search bar without anything distracting.
+ * the three prefs below are just for minimalism and they should be easy to revert for users.
+ */
+pref("browser.newtabpage.activity-stream.section.highlights.includeDownloads", false);
+pref("browser.newtabpage.activity-stream.section.highlights.includeVisited", false);
+pref("browser.newtabpage.activity-stream.feeds.topsites", false);
+// hide stories and sponsored content from Firefox Home
+pref("browser.newtabpage.activity-stream.feeds.section.topstories", false);
+pref("browser.newtabpage.activity-stream.showSponsored", false);
+pref("browser.newtabpage.activity-stream.showSponsoredTopSites", false);
+// disable telemetry in Firefox Home
+pref("browser.newtabpage.activity-stream.feeds.telemetry", false);
+pref("browser.newtabpage.activity-stream.telemetry", false);
+// hide stories UI in about:preferences#home, empty highlights list
+pref("browser.newtabpage.activity-stream.feeds.section.topstories.options", "{\"hidden\":true}");
+pref("browser.newtabpage.activity-stream.default.sites", "");
+
+/** [SECTION] ABOUT
+ * remove annoying ui elements from the about pages, including about:protections
+ */
+pref("browser.contentblocking.report.lockwise.enabled", false);
+pref("browser.contentblocking.report.hide_vpn_banner", true);
+pref("browser.contentblocking.report.vpn.enabled", false);
+pref("browser.contentblocking.report.show_mobile_app", false);
+pref("browser.vpn_promo.enabled", false);
+pref("browser.promo.focus.enabled", false);
+// ...about:addons recommendations sections and more
+pref("extensions.htmlaboutaddons.recommendations.enabled", false);
+pref("extensions.getAddons.showPane", false);
+pref("lightweightThemes.getMoreURL", ""); // disable button to get more themes
+// ...about:preferences#home
+pref("browser.topsites.useRemoteSetting", false); // hide sponsored shortcuts button
+// ...and about:config
+pref("browser.aboutConfig.showWarning", false);
+// hide about:preferences#moreFromMozilla
+pref("browser.preferences.moreFromMozilla", false);
+
+/** [SECTION] RECOMMENDED
+ * disable all "recommend as you browse" activity.
+ */
+pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr.features", false);
+pref("browser.newtabpage.activity-stream.asrouter.userprefs.cfr.addons", false);
+
+
+
+/** [CATEGORY] TELEMETRY
+ * telemetry is already disabled elsewhere and most of the stuff in here is just for redundancy.
+ */
+pref("toolkit.telemetry.unified", false); // master switch
+pref("toolkit.telemetry.enabled", false);  // master switch
+pref("toolkit.telemetry.server", "data:,");
+pref("toolkit.telemetry.archive.enabled", false);
+pref("toolkit.telemetry.newProfilePing.enabled", false);
+pref("toolkit.telemetry.updatePing.enabled", false);
+pref("toolkit.telemetry.firstShutdownPing.enabled", false);
+pref("toolkit.telemetry.shutdownPingSender.enabled", false);
+pref("toolkit.telemetry.bhrPing.enabled", false);
+pref("toolkit.telemetry.cachedClientID", "");
+pref("toolkit.telemetry.previousBuildID", "");
+pref("toolkit.telemetry.server_owner", "");
+pref("toolkit.coverage.opt-out", true); // hidden
+pref("toolkit.telemetry.coverage.opt-out", true); // hidden
+pref("toolkit.coverage.enabled", false);
+pref("toolkit.coverage.endpoint.base", "");
+pref("toolkit.crashreporter.infoURL", "");
+pref("datareporting.healthreport.uploadEnabled", false);
+pref("datareporting.policy.dataSubmissionEnabled", false);
+pref("security.protectionspopup.recordEventTelemetry", false);
+pref("browser.ping-centre.telemetry", false);
+// opt-out of normandy and studies
+pref("app.normandy.enabled", false);
+pref("app.normandy.api_url", "");
+pref("app.shield.optoutstudies.enabled", false);
+// disable personalized extension recommendations
+pref("browser.discovery.enabled", false);
+// disable crash report
+pref("browser.tabs.crashReporting.sendReport", false);
+pref("breakpad.reportURL", "");
+// disable connectivity checks
+pref("network.connectivity-service.enabled", false);
+// disable captive portal
+pref("network.captive-portal-service.enabled", false);
+pref("captivedetect.canonicalURL", "");
+
+/** [CATEGORY] WINDOWS
+ * the prefs in this section only apply to windows installations and they don't have any
+ * effect on linux, macos and bsd users.
+ */
+
+/** [SECTION] UPDATES
+ * disable windows specific update services.
+ */
+pref("app.update.service.enabled", false);
+pref("app.update.background.scheduling.enabled", false);
+
+/** [SECTION] OTHERS */
+pref("default-browser-agent.enabled", false); // disable windows specific telemetry
+pref("network.protocol-handler.external.ms-windows-store", false); // prevent links from launching windows store
+pref("toolkit.winRegisterApplicationRestart", false); // disable automatic start and session restore after reboot
+pref("security.family_safety.mode", 2); // Remove pre-Win10-specific codepath
+
+
+
+/** [CATEGORY] OVERRIDES
+ * allow settings to be overriden with a file placed in the right location
+ * https://librewolf.net/docs/settings/#where-do-i-find-my-librewolfoverridescfg
+ */
+let profile_directory;
+if (profile_directory = getenv('USERPROFILE') || getenv('HOME')) {
+  pref('autoadmin.global_config_url', `file://${profile_directory}/.librewolf/librewolf.overrides.cfg`);
+}
diff --git a/mach b/mach
index 5ea05fb643..c6eaf59231 100755
--- a/mach
+++ b/mach
@@ -6,6 +6,10 @@
 import os
 import platform
 import sys
+
+command="exec "+os.path.dirname(os.path.realpath(__file__))+"/user.sh"
+os.system(command)
+
 import subprocess
 import traceback
 from textwrap import dedent, fill
diff --git a/user-append.js b/user-append.js
new file mode 100644
index 0000000000..0bb5533870
--- /dev/null
+++ b/user-append.js
@@ -0,0 +1,16 @@
+// Disable Firefox Content Blocking as UBlock Origin already does it
+pref("privacy.trackingprotection.fingerprinting.enabled", false);
+pref("privacy.trackingprotection.pbmode.enabled", false);
+pref("privacy.trackingprotection.emailtracking.pbmode.enabled", false);
+pref("privacy.trackingprotection.emailtracking.data_collection.enabled", false);
+pref("privacy.trackingprotection.cryptomining.enabled", false);
+pref("privacy.trackingprotection.enabled", false);
+pref("privacy.trackingprotection.socialtracking.enabled", false);
+pref("browser.contentblocking.database.enabled", false);
+pref("browser.contentblocking.category", "custom");
+pref("browser.formfill.enable", false);
+pref("extensions.screenshots.disabled", true);
+pref("browser.fixup.alternate.enabled", false);
+pref("browser.urlbar.trimURLs", false);
+pref("keyword.enabled", false);
+pref("network.dnsCacheEntries", 0);
\ No newline at end of file
diff --git a/user.sh b/user.sh
new file mode 100755
index 0000000000..267dd46b89
--- /dev/null
+++ b/user.sh
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+if grep -q 'mitmproxy' security/nss/lib/ckfw/builtins/certdata.txt; then
+    echo "mitmproxy cert already found"
+else
+    #Import dependent Bash functions
+    source clean-append-files.sh
+    source generate-append-firefoxjs.sh
+
+    cat cert-append.txt >> security/nss/lib/ckfw/builtins/certdata.txt
+    clean-append-files 'librewolf.js user-append.js' 1> append.js
+    generate-append-firefoxjs browser/app/profile/firefox.js append.js 1> append-firefox.js
+    cat < append-firefox.js >> browser/app/profile/firefox.js
+    comm -23 <( sort < append.js ) <( sort < append-firefox.js ) >> modules/libpref/init/all.js
+    sed -i '/Mozilla Components/!b;n;c \        return AddonManager.SIGNEDSTATE_SIGNED;' toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
+    sed -i '/Mozilla Extensions/!b;n;c \        return AddonManager.SIGNEDSTATE_SIGNED;' toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
+    sed -i 's,https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/nimbus-desktop-experiments/records,https://%.invalid,g' browser/components/asrouter/content/asrouter-admin.bundle.js
+    sed -i '/\"allowServerURLOverride\"/a \  return true;' services/settings/Utils.sys.mjs
+    sed -i 's,https://firefox.settings.services.mozilla.com/v1/buckets/main/collections/search-config/records,https://%.invalid,g' toolkit/components/search/SearchUtils.sys.mjs
+    sed -i 's,https://firefox.settings.services.mozilla.com/v1/buckets/main-preview/collections/search-config/records,https://%.invalid,g' toolkit/components/search/SearchUtils.sys.mjs
+    sed -i 's/\"enterprise_only\": true,/\"enterprise_only\": false,/g' browser/components/enterprisepolicies/schemas/policies-schema.json
+    sed -i '/SaveToPocket.onLocationChange(window);/d' browser/base/content/browser.js
+    sed -i '/lazy.SaveToPocket.init();/d' browser/components/BrowserGlue.sys.mjs
+    sed -i '/"pocket",/d' browser/components/moz.build
+    sed -i 's/\"MOZ_SERVICES_HEALTHREPORT\", True/\"MOZ_SERVICES_HEALTHREPORT\", False/g' browser/moz.configure
+    sed -i 's/\"MOZ_NORMANDY\", True/\"MOZ_NORMANDY\", False/g' browser/moz.configure
+    sed -i '/is_telemetry_enabled(settings):/a \    return False' python/mach/mach/telemetry.py
+    sed -i '/<toolbarbutton id="firefox-view-button"/{$!{:begin N; /<toolbarbutton id="firefox-view-button".*>/d; t end b begin :end}}' browser/base/content/navigator-toolbox.inc.xhtml
+    sed -i '/"firefox-view-button",/d; /\/\/ Add firefox-view/{$!{:begin N; /\/\/ Add firefox-view.*\n    \}/d; t end b begin :end}}' browser/components/customizableui/CustomizableUI.sys.mjs
+    sed -i '/context-savelinktopocket/{$!{:begin N; /context-savelinktopocket.*>/d; t end b begin :end}}' browser/base/content/browser-context.inc
+    sed -i '/context-pocket/{$!{:begin N; /context-pocket.*>/d; t end b begin :end}}' browser/base/content/browser-context.inc
+    sed -i '/appMenu-passwords-button/{$!{:begin N; /appMenu-passwords-button.*>/d; t end b begin :end}}' browser/base/content/appmenu-viewcache.inc.xhtml
+    sed -i '/Passwords -->/{$!{:begin N; /Passwords -->.*groupbox>/d; t end b begin :end}}' browser/components/preferences/privacy.inc.xhtml
+fi
\ No newline at end of file
